src/frontend/magnify/src/components/profile/IdentityForm/IdentityForm.tsx

```diff
@@ -41,70 +33,41 @@ const messages = defineMessages({
   },
 });
 
+interface IdentityFormValues {
+  name: string;
+  username: string;
+  email: string;
+}
+
 export default function IdentityForm({ id, name, username, email }: IdentityFormProps) {
   const intl = useIntl();
-  const controller = useController();
-  const { mutate } = useMutation(controller.updateUser);
-
-  const { values, errors, modified, setValue, isValid, isModified } = useFormState(
-    { name, username, email },
-    {
-      name: validators(intl, requiredValidator),
-      username: validators(intl, requiredValidator, usernameValidator),
-      email: validators(intl, requiredValidator, emailValidator),
-    },
-  );
-
-  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const { value, name } = e.target;
-    setValue(name as 'name' | 'username' | 'email', value);
-  };
 
-  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    if (id) mutate({ id, ...values });
+  const handleSubmit = (values: IdentityFormValues) => {
+    console.log(id, values);
   };
 
   return (
-    <form onSubmit={handleSubmit}>
-      <TextField
-        label={intl.formatMessage(messages.nameLabel)}
-        name="name"
-        value={values.name}
-        errors={errors.name}
-        displayErrors={modified.name}
-        onChange={handleChange}
-        margin={{ bottom: 'small' }}
-        required
-      />
-      <TextField
-        label={intl.formatMessage(messages.usernameLabel)}
-        name="username"
-        value={values.username}
-        errors={errors.username}
-        displayErrors={modified.username}
-        onChange={handleChange}
-        margin={{ bottom: 'small' }}
-        required
-      />
-      <TextField
-        label={intl.formatMessage(messages.emailLabel)}
-        name="email"
-        value={values.email}
-        errors={errors.email}
-        displayErrors={modified.email}
-        onChange={handleChange}
-        margin={{ bottom: 'small' }}
-        required
-      />
-      <Box direction="row" justify="end" margin={{ top: 'small' }}>
-        <Button
-          primary
-          label={intl.formatMessage(messages.submitButtonLabel)}
-          disabled={!isModified || !isValid || !id}
-          type="submit"
-        />
-      </Box>
-    </form>
+    <Formik
+      validateOnChange={true}
+      initialValues={{ name, username, email }}
+      validationSchema={Yup.object().shape({
+        name: Yup.string().required(),
+        username: Yup.string()
+          .min(3, intl.formatMessage(validationMessages.usernameInvalid))
+          .max(16, intl.formatMessage(validationMessages.usernameInvalid))
+          .required(),
+        email: Yup.string().email().required(),
+      })}
```

Comment:
What do you think to move this into a dedicated variable

```suggestion
      validationSchema={validationSchema}
```